src/tests/Microsoft.PowerFx.Interpreter.Tests/PowerFxEvaluationTests.cs

```diff
@@ -336,10 +336,29 @@ protected override async Task<RunResult> RunAsyncInternal(string expr, string se
                     return new RunResult(newValue);
                 }
 
-                // Decimal TODO: This seems impossible without type information?  Float("1e300") will produce result that can't be serialized if treated as a decimal.
-                // Serialization test. Serialized expression must produce an identical result.
-                ParserOptions options = new ParserOptions() { NumberIsFloat = NumberIsFloat };
-                var newValueDeserialized = await engine.EvalAsync(newValue.ToExpression(), CancellationToken.None, options, runtimeConfig: runtimeConfig);
+                FormulaValue newValueDeserialized;
+
+                try
+                {
+                    // Serialization test. Serialized expression must produce an identical result.
+                    ParserOptions options = new ParserOptions() { NumberIsFloat = NumberIsFloat };
+                    newValueDeserialized = await engine.EvalAsync(newValue.ToExpression(), CancellationToken.None, options, runtimeConfig: runtimeConfig);
+                }
+                catch (InvalidOperationException e)
+                {
+                    // If we failed because of range limitations with decimal, retry with NumberIsFloat enabled
+                    // This is for tests that return 1e100 as a result, verifying proper floating point operation
+                    if (!NumberIsFloat && e.Message.Contains("value is too large"))
+                    {
+                        // Serialization test. Serialized expression must produce an identical result.
+                        ParserOptions options = new ParserOptions() { NumberIsFloat = true };
+                        newValueDeserialized = await engine.EvalAsync(newValue.ToExpression(), CancellationToken.None, options, runtimeConfig: runtimeConfig);
+                    }
+                    else
+                    {
+                        throw e;

```

Comment:
>e [](http://example.com/codeflow?start=30&length=1)

just "throw;"
adding "e" modifies the callstack - not sure if intended

Reply:
Done.